pkg/ebpf/tracer.go

```diff
@@ -656,6 +658,52 @@ func (t *Tracer) determineConnectionDirection(conn *ConnectionStats) ConnectionD
 	return OUTGOING
 }
 
+func (t *Tracer) determineConnectionIntraHost(connections []ConnectionStats) {
+
+	type connKey struct {
+		Address util.Address
+		Port    uint16
+		Type    ConnectionType
+	}
+
+	newConnKey := func(connStat *ConnectionStats, useRAddrAsKey bool) connKey {
+		key := connKey{Type: connStat.Type}
+		if useRAddrAsKey {
+			if connStat.IPTranslation == nil {
+				key.Address = connStat.Dest
+				key.Port = connStat.DPort
+			} else {
+				key.Address = connStat.IPTranslation.ReplSrcIP
+				key.Port = connStat.IPTranslation.ReplSrcPort
+			}
+		} else {
+			key.Address = connStat.Source
+			key.Port = connStat.SPort
+		}
+		return key
+	}
+
+	lAddrs := make(map[connKey]struct{}, 0)
```

Comment:
You can leave out the 0 here.

Reply:
Removed this - looks like it doesn't update here though. Resolving conversation.